src/CharterManager.sol

```diff
@@ -214,10 +214,10 @@ contract CharterManagerImp {
         ManagedGemJoinLike(gemJoin).join(getOrCreateProxy(usr), amt);
     }
 
-    function exit(address gemJoin, address usr, uint256 amt) external {
+    function exit(address gemJoin, address u, address usr, uint256 amt) external allowed(u) {
```

Comment:
I'm skeptical we should add this extra `u` parameter as it kind of breaks parity with the gem join interface. Is it necessary?

```
function exitGem(
    address gemJoin,
    uint256 amt
) external {
    // Exits token amount to the user's wallet as a token
    CharterLike(charter).exit(gemJoin, address(this), msg.sender, amt);
}
```

In the `exitGem` case above the tokens are being directly transferred to the user address, but the same can be achieved with the manager's flux similar to below.

```
function exitGem(
    address manager,
    address gemJoin,
    uint cdp,
    uint amt
) public {
    // Moves the amount from the CDP urn to proxy's address
    flux(manager, cdp, address(this), convertTo18(gemJoin, amt));

    // Exits token amount to the user's wallet as a token
    GemJoinLike(gemJoin).exit(msg.sender, amt);
}
```

Reply:
Yes, agreed. We should use the same technique and have the user flux before exit.
Removed this change.